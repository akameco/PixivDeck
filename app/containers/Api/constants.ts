import { API_REQUEST_SUCCESS_TYPE } from './actionTypes'

export const API_REQUEST_SUCCESS: API_REQUEST_SUCCESS_TYPE =
  'Api/API_REQUEST_SUCCESS'
