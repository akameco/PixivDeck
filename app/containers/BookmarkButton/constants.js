// @flow
import type {
  ADD_BOOKMARK_REQUEST_TYPE,
  ADD_BOOKMARK_SUCCESS_TYPE,
  ADD_BOOKMARK_FAILER_TYPE,
  DELETE_BOOKMARK_REQUEST_TYPE,
  DELETE_BOOKMARK_SUCCESS_TYPE,
  DELETE_BOOKMARK_FAILER_TYPE,
} from './actionTypes'

export const ADD_BOOKMARK_REQUEST: ADD_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ADD_BOOKMARK_REQUEST'
export const ADD_BOOKMARK_SUCCESS: ADD_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/ADD_BOOKMARK_SUCCESS'
export const ADD_BOOKMARK_FAILER: ADD_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/ADD_BOOKMARK_FAILER'
export const DELETE_BOOKMARK_REQUEST: DELETE_BOOKMARK_REQUEST_TYPE =
  'BookmarkButton/ELETE_BOOKMARK_REQUEST'
export const DELETE_BOOKMARK_SUCCESS: DELETE_BOOKMARK_SUCCESS_TYPE =
  'BookmarkButton/DELETE_BOOKMARK_SUCCESS'
export const DELETE_BOOKMARK_FAILER: DELETE_BOOKMARK_FAILER_TYPE =
  'BookmarkButton/DELETE_BOOKMARK_FAILER'
