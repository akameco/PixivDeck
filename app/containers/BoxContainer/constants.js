// @flow
import type { OPEN_PIXIV_TYPE } from './actionTypes'

export const OPEN_PIXIV: OPEN_PIXIV_TYPE = 'BoxContainer/OPEN_PIXIV'
