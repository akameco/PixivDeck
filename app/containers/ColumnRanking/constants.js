// @flow
import type {
  ADD_COLUMN_TYPE,
  ADD_COLUMN_SUCCESS_TYPE,
  SET_NEXT_URL_TYPE,
  CLERE_TYPE,
  FETCH_TYPE,
  FETCH_SUCCESS_TYPE,
  FETCH_FAILRE_TYPE,
  START_WATCH_TYPE,
  WATCH_NEW_TYPE,
} from './actionTypes'

export const ADD_COLUMN: ADD_COLUMN_TYPE = 'ColumnRanking/ADD_COLUMN'
export const ADD_COLUMN_SUCCESS: ADD_COLUMN_SUCCESS_TYPE =
  'ColumnRanking/ADD_COLUMN_SUCCESS'
export const SET_NEXT_URL: SET_NEXT_URL_TYPE = 'ColumnRanking/SET_NEXT_URL'
export const CLERE: CLERE_TYPE = 'ColumnRanking/CLERE'
export const FETCH: FETCH_TYPE = 'ColumnRanking/FETCH'
export const FETCH_SUCCESS: FETCH_SUCCESS_TYPE = 'ColumnRanking/FETCH_SUCCESS'
export const FETCH_FAILRE: FETCH_FAILRE_TYPE = 'ColumnRanking/FETCH_FAILRE'
export const START_WATCH: START_WATCH_TYPE = 'ColumnRanking/START_WATCH'
export const WATCH_NEW: WATCH_NEW_TYPE = 'ColumnRanking/WATCH_NEW'
