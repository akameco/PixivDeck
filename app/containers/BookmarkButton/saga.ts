import { call, put, takeEvery } from 'redux-saga/effects'
import * as api from 'containers/Api/sagas'
import { get } from '../Api/sagas'
import * as columnActions from '../ColumnBookmark/actions'
import * as Actions from './constants'
import * as actions from './actions'
import { Restrict } from './types'

interface Props {
  id: number
  restrict: Restrict
}
export function* bookmark({ id, restrict }: Props) {
  try {
    yield call(api.post, '/v2/illust/bookmark/add', {
      illustId: id,
      restrict,
    })
    yield put(actions.addBookmarkSuccess(id, restrict))
  } catch (error) {
    yield put(actions.addBookmarkFailer(id, error))
  }
}
export function* deleteTask({ id }: Props) {
  try {
    yield call(api.post, '/v1/illust/bookmark/delete', {
      illustId: id,
    })
    yield put(actions.deleteBookmarkSuccess(id))
    yield put(columnActions.removeItem('public', id))
  } catch (error) {
    yield put(actions.deleteBookmarkFailer(id, error))
  }
}

function* success({ id }: Props) {
  try {
    yield call(get, `/v1/illust/detail?illust_id=${id}`, true)
  } catch (error) {}
}

function* root() {
  yield takeEvery(Actions.ADD_BOOKMARK_REQUEST, bookmark)
  yield takeEvery(Actions.DELETE_BOOKMARK_REQUEST, deleteTask)
  yield takeEvery(
    [Actions.ADD_BOOKMARK_SUCCESS, Actions.DELETE_BOOKMARK_SUCCESS],
    success
  )
}

export default root
