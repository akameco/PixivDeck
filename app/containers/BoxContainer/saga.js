// @flow
import { shell } from 'electron'
import type { Saga } from 'redux-saga'
import { call, takeEvery } from 'redux-saga/effects'
import * as Actions from './constants'

function* openPixiv({ id }: { id: number }): Saga<void> {
  yield call(
    shell.openExternal,
    `http://www.pixiv.net/member_illust.php?mode=medium&illust_id=${id}`
  )
}

export default function* root(): Saga<void> {
  yield takeEvery(Actions.OPEN_PIXIV, openPixiv)
}
