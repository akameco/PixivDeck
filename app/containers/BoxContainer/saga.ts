import { shell } from 'electron'
import { call, takeEvery } from 'redux-saga/effects'
import * as Actions from './constants'

function* openPixiv({ id }: { id: number }) {
  yield call(
    shell.openExternal,
    `http://www.pixiv.net/member_illust.php?mode=medium&illust_id=${id}`
  )
}

export default function* root() {
  yield takeEvery(Actions.OPEN_PIXIV, openPixiv)
}
