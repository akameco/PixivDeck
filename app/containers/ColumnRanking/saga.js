// @flow
import { type Saga } from 'redux-saga'
import {
  put,
  select,
  call,
  takeEvery,
  fork,
  take,
  delay,
} from 'redux-saga/effects'
import { addTable } from 'containers/ColumnManager/actions'
import * as api from '../Column/sagas'
import * as Actions from './constants'
import * as actions from './actions'
import type { Mode } from './reducer'
import * as selectors from './selectors'

type Action = { id: Mode }

export function* addColumn({ id }: Action): Saga<void> {
  const modes: Array<?Mode> = yield select(selectors.makeSelectModes())
  if (modes.every(v => v !== id)) {
    yield put(actions.addColumnSuccess(id))
  }

  yield put(addTable(`ranking-${id}`, { columnId: id, type: 'RANKING' }))
}

const getEndpoint = id => `/v1/illust/ranking?mode=${id}`

export function* fetch(action: Action): Saga<void> {
  const { ids, nextUrl } = yield select(selectors.makeSelectColumn(), action)
  const endpoint = nextUrl ? nextUrl : getEndpoint(action.id)
  yield call(api.fetchColumn, endpoint, action.id, actions, ids)
}

export function* fetchNew({ id }: Action): Saga<void> {
  const { ids } = yield select(selectors.makeSelectColumn(), { id })
  yield call(
    api.fetchNew,
    { endpoint: getEndpoint(id), id, ids, order: 'overwrite' },
    actions
  )
}

// $FlowFixMe
export function* watchNewIllust(): Saga<*> {
  while (true) {
    const { id } = yield take(Actions.START_WATCH)
    const interval = yield select(selectors.getInterval, { id })
    yield delay(interval)
    yield call(fetchNew, { id })
    yield put(actions.watchNew(id))
  }
}

export default function* root(): Saga<void> {
  yield takeEvery(Actions.ADD_COLUMN, addColumn)
  yield takeEvery(Actions.FETCH, fetch)
  yield fork(watchNewIllust)
}
