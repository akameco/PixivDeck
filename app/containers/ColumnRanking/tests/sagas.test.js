// @flow
import { takeEvery, take, call } from 'redux-saga/effects'
import * as sagas from '../saga'
import * as constants from '../constants'

test('root', () => {
  const gen = sagas.default()
  expect(gen.next().value).toStrictEqual(
    takeEvery(constants.ADD_COLUMN, sagas.addColumn)
  )
  expect(gen.next().value).toStrictEqual(
    takeEvery(constants.FETCH, sagas.fetch)
  )
  expect(gen.next().value).toMatchSnapshot()
})

test('watchNewIllust', () => {
  const gen = sagas.watchNewIllust()
  expect(gen.next().value).toStrictEqual(take(constants.START_WATCH))
  expect(gen.next({ id: 1 }).value).toMatchSnapshot()
  expect(gen.next().value).toMatchSnapshot()
  // $FlowFixMe
  expect(gen.next().value).toStrictEqual(call(sagas.fetchNew, { id: 1 }))
})
