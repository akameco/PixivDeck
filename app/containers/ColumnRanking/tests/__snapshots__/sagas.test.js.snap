// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`root 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
  "type": "FORK",
}
`;

exports[`watchNewIllust 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      Object {
        "id": 1,
      },
    ],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`watchNewIllust 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      undefined,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
